python/ray/autoscaler/v2/tests/test_instance_manager.py

```diff
@@ -0,0 +1,172 @@
+import os
+import sys
+import unittest
+
+# coding: utf-8
+from typing import Dict, Set
+from unittest.mock import patch
+
+import pytest
+
+from ray.autoscaler.v2.instance_manager.instance_manager import InstanceUtil
+from ray.core.generated.instance_manager_pb2 import Instance
+
+
+def exists_path(
+    src: Instance.InstanceStatus,
+    dst: Instance.InstanceStatus,
+    graphs: Dict["Instance.InstanceStatus", Set["Instance.InstanceStatus"]],
+) -> bool:
+    # BFS search from src to dst to see if there is a path
+    # from src to dst. There's no path if src == dst.
+    visited = set()
+    queue = [src]
+    if src == dst:
+        return False
+    while queue:
+        node = queue.pop(0)
+        if node not in visited:
+            visited.add(node)
+            queue.extend(graphs[node])
+    return dst in visited
+
+
+class InstanceUtilTest(unittest.TestCase):
+    def test_transition_graph(self):
```

Comment:
add a test to verify is_ray_running_reachable is correct based on the graph?

Reply:
yes - it's in `test_is_ray_running_reachable`